examples/pytorch/language-modeling/run_clm.py

```diff
@@ -433,14 +442,18 @@ def tokenize_function(examples):
         return output
 
     with training_args.main_process_first(desc="dataset map tokenization"):
-        tokenized_datasets = raw_datasets.map(
-            tokenize_function,
+        kwargs = dict(
             batched=True,
-            num_proc=data_args.preprocessing_num_workers,
             remove_columns=column_names,
-            load_from_cache_file=not data_args.overwrite_cache,
-            desc="Running tokenizer on dataset",
         )
+        if not data_args.streaming:
+            kwargs.update(
+                num_proc=data_args.preprocessing_num_workers,
+                load_from_cache_file=not data_args.overwrite_cache,
+                desc="Running tokenizer on dataset",
+            )
+
+        tokenized_datasets = raw_datasets.map(tokenize_function, **kwargs)
```

Comment:
Same here, let's not refactor kwars. I prefer seeing two different calls to the map function.

Reply:
same problem here, `map` on `IterableDaset` doesn't expect the same args.